geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionAdvisor.java

```diff
@@ -1314,7 +1320,7 @@ public Set adviseProfileRemove() {
   private synchronized boolean basicAddProfile(Profile p) {
     // don't add more than once, but replace existing profile
     // try {
-
+    profilesVersion++;
```

Comment:
You are safe here because the method is synchronized but `volatile` increment operations are not atomic.